MapboxVisionTests/VisionManagerTests.swift

```diff
@@ -0,0 +1,139 @@
+@testable import MapboxVision
+import XCTest
+
+class VisionManagerTests: XCTestCase {
+    var visionManager: VisionManager?
+    var dependencies: VisionDependencies?
+
+    override func setUp() {
+        super.setUp()
+
+        let native = MockNative()
+
+        dependencies = VisionDependencies(
+            native: native,
+            synchronizer: MockSynchronizable(),
+            recorder: MockSessionRecorder(),
+            dataProvider: MockDataProvider(),
+            deviceInfo: DeviceInfoProvider()
+        )
+
+        self.visionManager = VisionManager(dependencies: dependencies!, videoSource: MockVideoSource())
+    }
+
+    func test_AssertOnVisionManagerDeallocWithoutDestroy() {
+        // Description:
+        // We called destroy in destructors of VisionManager, BaseVisionManager, VisionManagerNative and VisionManagerBaseNative,
+        // so VisionManagerBaseNative's destroy method called multiple times, so we tried to destroy already destroyed resources,
+        // which caused an EXC_BAD_ACCESS.
+        //
+        // Nevertheless the logic was sort of right, it should do assertFailure on deinit without destroy was called
+        //
+        // Given
+        // object of VisionManager
+        // When
+        // we deallocate it without destroy
+        // Then
+        // Should be assertion failure
+        MVTAssertAssertionFailure(
+            self.visionManager = nil,
+            "VisionManager should call assertioFailure when we release the object without calling destroy"
+        )
+    }
+
+    func test_NoAssertOnVisionManagerDeallocAfterDestroy() {
+        // Given
+        // object of VisionManager
+
+        // When
+        // we call destroy and release the instance
+        self.visionManager?.destroy()
+
+        // Then
+        // Shouldn't be assertion failure
+        MVTAssertNoAssertionFailure(
+            self.visionManager = nil,
+            "VisionManager should be successfully deallocated after calling destroy and releasing the object"
+        )
+    }
+
+    func test_RecordingPermissions() {
+        // Given
+        // VisionManager with default country
+
+        guard let visionManager = self.visionManager, let recorder = dependencies?.recorder as? MockSessionRecorder else {
+            XCTAssert(false)
```

Comment:
Please use XCTFail and include description in assert.

Reply:
done